src/main.rs

```diff
@@ -130,144 +136,144 @@ fn main() {
+                    };
+                    // Somehow using println! leads to the binary panicking
+                    // when its output is piped.
+                    // So, we're handling a Broken Pipe error and exiting with 0 anyway
+                    let stdout = std::io::stdout();
+                    {
+                        let mut handle = stdout.lock();
+                        handle.write_all(line.as_bytes()).unwrap_or_else(|e| {
+                            match e.kind() {
+                                std::io::ErrorKind::BrokenPipe => std::process::exit(0),
+                                _ => std::process::exit(1),
+                            };
+                        });
+                    }
                 }
-            }
-        });
-        let percentage_progress = exercises_done as f32 / exercises.len() as f32 * 100.0;
-        println!(
-            "Progress: You completed {} / {} exercises ({:.2} %).",
-            exercises_done,
-            exercises.len(),
-            percentage_progress
-        );
-        std::process::exit(0);
-    }
-
-    // Handle the run command
-    if let Some(ref matches) = matches.subcommand_matches("run") {
-        let name = matches.value_of("name").unwrap();
-
-        let matching_exercise = |e: &&Exercise| name == e.name;
+            });
+            let percentage_progress = exercises_done as f32 / exercises.len() as f32 * 100.0;
+            println!(
+                "Progress: You completed {} / {} exercises ({:.2} %).",
+                exercises_done,
+                exercises.len(),
+                percentage_progress
+            );
+            std::process::exit(0);
+        }
 
-        let exercise = exercises.iter().find(matching_exercise).unwrap_or_else(|| {
-            println!("No exercise found for your given name!");
-            std::process::exit(1)
-        });
+        Subcommands::Run(subargs) => {
+            let name = subargs.name;
 
-        run(&exercise, verbose).unwrap_or_else(|_| std::process::exit(1));
-    }
+            let matching_exercise = |e: &&Exercise| name == e.name;
 
-    if let Some(ref matches) = matches.subcommand_matches("hint") {
-        let name = matches.value_of("name").unwrap();
-
-        let exercise = exercises
-            .iter()
-            .find(|e| name == e.name)
-            .unwrap_or_else(|| {
+            let exercise = exercises.iter().find(matching_exercise).unwrap_or_else(|| {
                 println!("No exercise found for your given name!");
                 std::process::exit(1)
             });
 
-        println!("{}", exercise.hint);
-    }
+            run(&exercise, verbose).unwrap_or_else(|_| std::process::exit(1));
+        }
 
-    // Handle the verify command
-    if matches.subcommand_matches("verify").is_some() {
-        verify(&exercises, verbose).unwrap_or_else(|_| std::process::exit(1));
-    }
+        Subcommands::Hint(subargs) => {
```

Comment:
Carryover from previous commit but `Subcommands::Run` and `Subcommands::Hint` repeat the same logic.